src/utils/mappingUtil.ts

```diff
@@ -124,6 +124,8 @@ export class MappingUtil {
     testResult.testTypes.forEach((testType: models.TestType) => {
       if (enums.SPECIALIST_TEST_TYPE_IDS.includes(testType.testTypeId)) {
         testType.defects = [];
+      } else {
+        testType.ivaDefects = [];
```

Comment:
I'm still not clear why we need to do this. Are we sure that we want to always clear the defect array down?

Reply:
Yes i'm not sure on this one either. Currently, if the test is a specialist test that has an ID included in that enum then it wipes the testType.defects array (overwrites with [ ]).

Converesely, now we are adding the ivaDefects array, if the test type id is not in that enum, should we be 'cleaning' that up too?

Reply:
are we sure that all of the test types in that enum map to IVA? do we need a more specific guard on here?

Reply:
I think that we need a more specific guard as those id's include more than just IVA

Reply:
The logic to wipe the testType.defects is already in the code and has been for a number of years, I wonder if this was done because the defects array is mandatory? Also I believe this code only runs when a test-result is updated.

With regards to testType.ivaDefects, as per my comment in the schema, should that actually be mandatory as is above? I'm not sure it should, given this should only ever be populated for an IVA30?